test/minitest/test_time_reporter.rb

```diff
@@ -0,0 +1,71 @@
+# encoding: utf-8
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+# frozen_string_literal: true
+
+MINITEST_CLASS_NAME = if defined?(MiniTest::VERSION)
+  MiniTest::StatisticsReporter
+elsif defined?(Minitest::VERSION)
+  Minitest::StatisticsReporter
+else
+  class DummyClass; end
+  DummyClass
+end
+
+class TestTimeReporter < MINITEST_CLASS_NAME
+  attr_accessor :test_times
+
+  # Keep the value for this constant the same as in Multiverse::Runner#run
+  TEST_TIME_REPORT = File.join(File.expand_path(File.dirname(__FILE__)), 'minitest_time_report')
+
+  def initialize(options = {})
+    super
+    File.open(TEST_TIME_REPORT, 'w+') unless File.exist?(TEST_TIME_REPORT)
```

Comment:
I'm concerned about open file handles being left dandling. Is this intended to simply "touch" a new file which will be appended to later on?

If so, is `File.open(filename, 'w+')` equivalent to `file = File.open(filename, 'w+'); file.close`? I just want to make sure we only have a handle open while we're actively writing.

Also if this is truly just a "touch" operation, there's `FileUtils.touch`.

Reply:
@fallwith  - `File.open` automatically closes the file when a block is passed to it and that block completes, so it wouldn't be closed in this case. I'll adjust to using `touch` instead. Thanks!

Reply:
ok, cool. using touch will definitely not even open a handle, so that's perfect.

in my testing, I was worried about `File.open`:

```ruby
> File.open('test', 'w').closed?
=> false

> file = File.open('test', 'w')
> file.close
> file.closed?
=> true
```

it was bugging me that the first example returned `false`

Reply:
Resolved in: [2c9bf39](https://github.com/newrelic/newrelic-ruby-agent/pull/1468/commits/2c9bf39904a1dd9ed964e9de6d0b84abd224ad8a)